google/cloud/storage/bucket.py

```diff
@@ -1263,6 +1285,102 @@ def list_blobs(
         iterator.prefixes = set()
         return iterator
 
+    def list_prefixes(
```

Comment:
Pending feedback from issue filer. I'm not convinced this is the right path forward.